common/djangoapps/student/views.py

```diff
@@ -2717,3 +2717,23 @@ def get_context_data(self, **kwargs):
         })
 
         return context
+
+
+class LogoutRedirectView(TemplateView):
+    """
+    Logs out user, performing the same logic as LogoutView, and redirects the user to a specified target.
+    """
+
+    template_name = 'logout.html'
+
+    def get_context_data(self, **kwargs):
+        context = super(LogoutRedirectView, self).get_context_data(**kwargs)
+        target = kwargs.get('redirect_to', '/')
+        logout_uri = "{}?{}".format(reverse('logout'), 'no_redirect=1')
+
+        context.update({
+            'target': target,
+            'logout_uri': logout_uri,
```

Comment:
It looks like the existing template needs this to be an iterable called `logout_uris` containing a number of items? So just make this a list containing the single URI you need.